source/store.go

```diff
@@ -208,7 +209,18 @@ func BuildWithConfig(ctx context.Context, source string, p ClientGenerator, cfg
 		if err != nil {
 			return nil, err
 		}
-		return NewNodeSource(ctx, client, cfg.AnnotationFilter, cfg.FQDNTemplate)
+		nodeLabelFilter := labels.Everything()
+		if requirements, selectable := cfg.LabelFilter.Requirements(); selectable {
+			for _, requirement := range requirements {
+				nodeLabelFilter = nodeLabelFilter.Add(requirement)
+			}
+		}
+		if requirements, selectable := cfg.NodeLabelFilter.Requirements(); selectable {
+			for _, requirement := range requirements {
+				nodeLabelFilter = nodeLabelFilter.Add(requirement)
+			}
+		}
+		return NewNodeSource(ctx, client, cfg.AnnotationFilter, cfg.FQDNTemplate, nodeLabelFilter)
```

Comment:
I thought more of the mutual exclusion of these 2 flags. With this implementation it's only possible to add up to the generic label filter and impossible to specify a different filter for the node source. How about using only `--node-label-filter`if it's there and default to `--label-filter` if it's not?